dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/DefaultTransformStrategy.java

```diff
@@ -0,0 +1,341 @@
+                .put("languageFlag", getLiteralLocale(language.getLanguageCode(), language.getCountryCode()));
+
+        final String iso = UtilMethods.isSet(language.getCountryCode())
+                ? language.getLanguageCode() + StringPool.DASH + language.getCountryCode()
+                : language.getLanguageCode();
+        builder.put("isoCode", iso.toLowerCase());
+
+        if(wrapAsMap){
+            builder.put("id", language.getId());
+            return ImmutableMap.of("languageMap", builder.build(), "language",language.getLanguage());
+        }
+        return builder.build();
+    }
+
+
+    /**
+     * Constant fields are added down here
+     * @param contentlet
+     * @param map
+     * @param options
+     */
+    private void addConstants(final Contentlet contentlet, final Map<String, Object> map, final Set<TransformOptions> options){
+          if(!options.contains(CONSTANTS)){
+             return;
+          }
+            contentlet.getContentType().fields(ConstantField.class)
+                    .stream()
+                    .filter(Objects::nonNull)
+                    .forEach(f-> map.put(f.variable(), f.values()));
+    }
+
+
+    /**
+     * This method includes binaries in the resulting view if so is indicated
+     * After the execution of this method if the BINARIES flag is turned on
+     * all the binary fields will be replaced and transformed by a /dA/.. path.
+     */
+    private void addBinaries(final Contentlet contentlet, final Map<String, Object> map, final Set<TransformOptions> options) {
+
+        final List<Field> binaries = contentlet.getContentType().fields(BinaryField.class);
+
+        if (binaries.isEmpty()) {
+            return;
+        }
+
+        //If we dont want any binaries making it into the final map
+        if(options.contains(FILTER_BINARIES)){
+            binaries.forEach(field -> {
+                map.remove(field.variable());
+            });
+            Logger.info(DefaultTransformStrategy.class, ()->"Transformer was instructed to exclude binaries.");
+            return;
+        }
+
+        // if we want to include binaries as they are (java.io.File) this is the flag you should turn on.
+        if (options.contains(BINARIES)) {
+            for (final Field field : binaries) {
+                try {
+                    final String velocityVarName = field.variable();
+                    //Extra precaution in case we are attempting to process a contentlet that has already been transformed.
+                    if (map.get(velocityVarName) instanceof File) {
+                        final File conBinary = contentlet.getBinary(field.variable());
+                        if (null != conBinary) {
+                            //If we want to see binaries. The binary-field per se. Must be replaced by file-name. We dont want to disclose any file specifics.
+
+                            if (conBinary.exists()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)